ports/mimxrt/machine_i2c.c

```diff
@@ -82,24 +84,26 @@ bool lpi2c_set_iomux(int8_t hw_i2c, uint8_t drive) {
 
 static void machine_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     machine_i2c_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    mp_printf(print, "I2C(%u, freq=%u)",
-        self->i2c_id, self->master_config->baudRate_Hz);
+    mp_printf(print, "I2C(%u, freq=%u, timeout=%u)",
+        self->i2c_id, self->master_config->baudRate_Hz,
+        self->master_config->pinLowTimeout_ns / 1000000);
 }
 
 mp_obj_t machine_i2c_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
-    enum { ARG_id, ARG_freq, ARG_drive};
+    enum { ARG_id, ARG_freq, ARG_drive, ARG_timeout};
     static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_id, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_id, MP_ARG_INT, {.u_int = DEFAULT_I2C_ID} },
         { MP_QSTR_freq, MP_ARG_INT, {.u_int = DEFAULT_I2C_FREQ} },
         { MP_QSTR_drive, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = DEFAULT_I2C_DRIVE} },
+        { MP_QSTR_timeout, MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = DEFAULT_I2C_TIMEOUT} },
```

Comment:
stm32, rp2 and the common `machine.SoftI2C` have this timeout parameter measured in microseconds, and default it to 50ms.  So please follow that convention here as well.

The name of this kwarg should stay as `timeout`, but I suggest calling the default macro `DEFAULT_I2C_TIMEOUT_US` to be clear about units (and set it to 50000).

Then adjust the logic converting `pinLowTimeout_ns` to/from this timeout value, to use 1000 instead of 1000000.